File: android/app/src/main/java/com/noop/analytics/DailySignalStatus.kt
package com.noop.analytics

/**
 * Conservative presentation state for the Daily Signal header.
 *
 * This combines existing engine outputs. It does not calculate another score and never turns missing
 * or thin data into a reassuring green state.
 */
enum class DailySignalStatus {
    BUILDING,
    STEADY,
    WATCH,
    ALERT;

    companion object {
        fun resolve(
            readiness: ReadinessEngine.Readiness?,
            illness: IllnessSignalEngine.Result?,
        ): DailySignalStatus {
            when (illness?.displayState) {
                IllnessSignalEngine.DisplayState.ALERT -> return ALERT
                IllnessSignalEngine.DisplayState.WATCH -> return WATCH
                IllnessSignalEngine.DisplayState.BUILDING -> return BUILDING
                IllnessSignalEngine.DisplayState.STEADY, null -> Unit
            }

            if (readiness == null || readiness.confidence != ScoreConfidence.SOLID) return BUILDING
            return when (readiness.level) {
                ReadinessEngine.Level.PRIMED, ReadinessEngine.Level.BALANCED -> STEADY
                ReadinessEngine.Level.STRAINED, ReadinessEngine.Level.RUNDOWN -> WATCH
                ReadinessEngine.Level.INSUFFICIENT -> BUILDING
            }
        }
    }
}

File: Packages/StrandAnalytics/Sources/StrandAnalytics/DailySignalStatus.swift
import Foundation

/// One conservative state for the Daily Signal header.
///
/// This is presentation policy, not another health formula. It combines the existing readiness and
/// multi-signal results without upgrading missing or thin data into a reassuring green state.
public enum DailySignalStatus: String, Equatable, Sendable, Codable {
    case building
    case steady
    case watch
    case alert

    public static func resolve(
        readiness: ReadinessEngine.Readiness?,
        illness: IllnessSignalEngine.Result?
    ) -> DailySignalStatus {
        if let illness {
            switch illness.displayState {
            case .alert:
                return .alert
            case .watch:
                return .watch
            case .building:
                return .building
            case .steady:
                break
            }
        }

        guard let readiness, readiness.confidence == .solid else { return .building }
        switch readiness.level {
        case .primed, .balanced:
            return .steady
        case .strained, .rundown:
            return .watch
        case .insufficient:
            return .building
        }
    }
}
